src/policy/lockfreeimmortalspace.rs

```diff
@@ -108,15 +107,23 @@ impl<VM: VMBinding> Space<VM> for LockFreeImmortalSpace<VM> {
     }
 
     fn reserved_pages(&self) -> usize {
-        let cursor = unsafe { Address::from_usize(self.cursor.load(Ordering::Relaxed)) };
+        let cursor = self.cursor.load(Ordering::Relaxed);
         let data_pages = conversions::bytes_to_pages_up(self.limit - cursor);
         let meta_pages = self.metadata.calculate_reserved_pages(data_pages);
         data_pages + meta_pages
     }
 
     fn acquire(&self, _tls: VMThread, pages: usize) -> Address {
         let bytes = conversions::pages_to_bytes(pages);
-        let start = unsafe { Address::from_usize(self.cursor.fetch_add(bytes, Ordering::Relaxed)) };
+        let start = self
+            .cursor
+            .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |addr| {
+                Some(addr.add(ByteSize::from(bytes)))
+            });
```

Comment:
Just a minor point: you can just do `unwrap()` here. As the closure always returns `Some`, we do not expect to see a `Err` value from `fetch_update`, thus error handling is unnecessary.

Reply:
I'd prefer using `expect` and it is a better practice comparing to matching